backend/pkg/logger/logger.go

```diff
@@ -0,0 +1,122 @@
+package logger
+
+import (
+	"encoding/csv"
+	"fmt"
+	"os"
+	"sync"
+	"time"
+
+	"github.com/HyperloopUPV-H8/h9-backend/pkg/abstraction"
+)
+
+var _ abstraction.Logger = &Logger{}
+
+type Record struct {
+	NameValue abstraction.LoggerName
+	Data      interface{}
+	Timestamp time.Time
+}
+
+// Type Request
+//TODO!
+
+func (r *Record) Name() abstraction.LoggerName {
+	return r.NameValue
+}
+
+type Logger struct {
+	buffer        []Record
+	flushInterval time.Duration
+	mu            sync.Mutex
+	ticker        *time.Ticker
+	running       bool
+}
+
+func NewLogger(flushInterval time.Duration) *Logger {
+	if flushInterval == 0 {
+		flushInterval = 5 * time.Second
+	}
+	return &Logger{
+		flushInterval: flushInterval,
+	}
+}
+
+func (l *Logger) Start() {
+	l.mu.Lock()
+	defer l.mu.Unlock()
+	if l.running {
+		return
+	}
+
+	l.running = true
+	l.ticker = time.NewTicker(l.flushInterval)
+
+	go func() {
+		for {
+			select {
+			case <-l.ticker.C:
+				l.flush()
+			}
+		}
+	}()
```

Comment:
return from this function after the ticker is stopped because otherwise you will have each time a new ticker in memory that won't be freed because this goroutine references it all the time